internal/authorization/middlewares.go

```diff
@@ -166,10 +130,11 @@ func (mdw *Middleware) Authorize() func(http.Handler) http.Handler {
 					return
 				}
 
-				user := mdw.user(r)
-				isAdmin := mdw.admin(r)
+				// if we got here then `principal` must be != nil
+				principal := authentication.PrincipalFromContext(r.Context())
```

Comment:
I think it would make sense, even at the beginning of this.
Right now the panic would still happend due to invoking a method on a nil object, but an explicit panic would be way clearer. I'll add it, thanks!

Reply:
we wouldnt recognize this as an error until someone logs in, which potentially might be after days

if we panic it should be at bootstrap